### PR TITLE
[SW-1787] Obtaining nodes can be done via rest API in all cases on External backend

### PR DESCRIPTION
**Pull request by jakubhava**
We now require web enabled on external backend and this simplifies a code a little bit

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendUtils.scala [hunk 1]
```diff
@@ -17,28 +17,97 @@
...
+import water.api.RestAPIManager
+import water.util.Log
+import water.{ExternalFrameUtils, H2O, H2OStarter, Paxos}
 
 private[backends] trait ExternalBackendUtils extends SharedBackendUtils {
 
+  protected def startH2OClient(hc: H2OContext, nodes: Array[NodeDesc]): Unit = {
+    val conf = hc.getConf
+    val args = ExternalH2OBackend.getH2OClientArgs(conf).toArray
+    val launcherArgs = ExternalH2OBackend.toH2OArgs(args, nodes)
```
Comment (jakubhava) [inline core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendUtils.scala]: here we pass the nodes to flatfile

File: core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalH2OBackend.scala [hunk 2]
```diff
@@ -85,18 +82,13 @@ class ExternalH2OBackend(val hc: H2OContext) extends SparklingBackend with Loggi
...
+      "-J", "-rest_api_ping_timeout", "-J", conf.clientCheckRetryTimeout.toString
     )
 
-    if (isRestApiBasedClient(hc)) {
-      cmdToLaunch = cmdToLaunch ++ Seq[String]("-J", "-rest_api_ping_timeout", "-J", conf.clientCheckRetryTimeout.toString)
-    } else {
+    if (!isRestApiBasedClient(hc)) {
       cmdToLaunch = cmdToLaunch ++ Seq[String](
-        "-J", "-client_disconnect_timeout", "-J", conf.clientCheckRetryTimeout.toString,
-        "-J", "-watchdog_stop_without_client",
```
Comment (jakubhava) [inline core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalH2OBackend.scala]: no need for watchdog, we have the rest api check we can use also in regular client mode of external backend

File: core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendConf.scala [hunk 3]
```diff
@@ -90,8 +90,12 @@ trait ExternalBackendConf extends SharedBackendConf {
   }
 
   def setClusterSize(clusterSize: Int) = set(PROP_EXTERNAL_CLUSTER_SIZE._1, clusterSize.toString)
-
-  def setClientConnectionTimeout(timeout: Int) = set(PROP_EXTERNAL_CLIENT_CONNECTION_TIMEOUT._1, timeout.toString)
+  @DeprecatedMethod
+  def setClientConnectionTimeout(timeout: Int) = {
+    logWarning(s"Setting the ${PROP_EXTERNAL_CLIENT_CONNECTION_TIMEOUT._1} has not effect and is not required" +
```
Comment (mn-mikke) [inline core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendConf.scala]: nit: Shouldn't the method name be mentioned in the warning message?

File: core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendUtils.scala [hunk 4]
```diff
@@ -17,49 +17,100 @@
...
+             |External H2O cluster is not running or could not be connected to. Provided configuration:
+             |  cluster name            : ${conf.cloudName.get}
+             |  cluster representative  : ${conf.h2oCluster.getOrElse("Using multi-cast discovery!")}
+             |  cluster start timeout   : ${conf.clusterStartTimeout} sec
+             |
+             |It is possible that in case you provided only the cluster name, h2o is not able to cloud up
+             |because multi-cast communication is limited in your network. In that case, please consider starting the
+             |external H2O cluster with flatfile and set the following configuration '${
+            ExternalBackendConf.
+              PROP_EXTERNAL_CLUSTER_REPRESENTATIVE._1
```
Comment (mn-mikke) [inline core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendUtils.scala]: nit: `ExternalBackendConf.PROP_EXTERNAL_CLUSTER_REPRESENTATIVE._1` could be on one line.

File: core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendConf.scala (hunk 3 above)
Comment (jakubhava) [reply core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendConf.scala]: 👍

File: core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendUtils.scala (hunk 4 above)
Comment (jakubhava) [reply core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendUtils.scala]: 👍

**Pull request merged**